File: www/stack_wasm.js
// Mock WebAssembly module for testing
function StackModule() {
    // Create a simple stack implementation in JavaScript
    let elements = [];
    const maxSize = 10;

    return Promise.resolve({
        isEmpty: function() {
            return elements.length === 0;
        },
        isFull: function() {
            return elements.length >= maxSize;
        },
        push: function(value) {
            if (elements.length >= maxSize) {
                return false;
            }
            elements.push(value);
            return true;
        },
        pop: function() {
            if (elements.length === 0) {
                return -1;
            }
            return elements.pop();
        },
        peek: function() {
            if (elements.length === 0) {
                return -1;
            }
            return elements[elements.length - 1];
        },
        size: function() {
            return elements.length;
        },
        maxSize: function() {
            return maxSize;
        },
        getElements: function() {
            return elements;
        }
    });
} 

File: src/stack_wasm.cpp
#include <emscripten/bind.h>
#include <emscripten/emscripten.h>
#include "../include/Stack.h"
#include <vector>
#include <string>

Stack stack(10);

bool isEmpty() {
    return stack.isEmpty();
}

bool isFull() {
    return stack.isFull();
}

bool push(int value) {
    try {
        stack.push(value);
        return true;
    } catch (const std::exception& e) {
        return false;
    }
}

int pop() {
    try {
        return stack.pop();
    } catch (const std::exception& e) {
        return -1;
    }
}

int peek() {
    try {
        return stack.peek();
    } catch (const std::exception& e) {
        return -1;
    }
}

size_t size() {
    return stack.size();
}

size_t maxSize() {
    return stack.getMaxSize();
}

emscripten::val getElements() {
    std::vector<int> elements = stack.getElements();
    return emscripten::val(emscripten::typed_memory_view(
        elements.size(),
        elements.data()
    ));
}

EMSCRIPTEN_BINDINGS(stack_module) {
    emscripten::function("isEmpty", &isEmpty);
    emscripten::function("isFull", &isFull);
    emscripten::function("push", &push);
    emscripten::function("pop", &pop);
    emscripten::function("peek", &peek);
    emscripten::function("size", &size);
    emscripten::function("maxSize", &maxSize);
    emscripten::function("getElements", &getElements);
}
